app/components/agenda/agenda-header/agenda-version-actions.js

```diff
@@ -342,22 +342,25 @@ export default class AgendaAgendaHeaderAgendaVersionActions extends Component {
         undefined,
         this.intl.t('an-agenda-was-approved-since-modal-was-opened')
       );
-      const agendaitemsNotOk = await this.allAgendaitemsNotOk();
       const newAgendaId = await approveDesignAgenda(this.args.currentAgenda);
       const newAgenda = await this.store.findRecord('agenda', newAgendaId);
+      const agendaCheckMappingsMap = new Map(
+        this.agendaCheckMapping.map(({ uri, generatedName }) => [uri, generatedName])
+      );
       await this.documentService.setGeneratedPieceNames(
         this.args.currentAgenda.id,
-        this.agendaCheckMapping,
+        agendaCheckMappingsMap,
         this.openAgendaCheckTimestamp,
       );
       await this.documentService.stampDocumentsOfAgenda(this.args.currentAgenda.id);
       // Data reloading
       await this.reloadAgenda(this.args.currentAgenda);
       await this.reloadAgendaitemsOfAgenda(this.args.currentAgenda);
       await this.reloadMeeting();
-      if (agendaitemsNotOk?.length) {
```

Comment:
`if (agendaitemsNotOk?.length) {`
this was used to fix gaps in numbering if a new agendaitem was removed during approval.
After first refactor > always regenerate everything, this removal was fine.
After second refactor (only regen those with new documents be name (not BIS)), this is no longer ok